_posts/2016-05-05-cargo-pillars.md

```diff
@@ -0,0 +1,489 @@
+---
+layout: post
+title: "Cargo: predictable dependency management"
+author: Yehuda Katz
+description: "Cargo makes dependency management in Rust easy and predictable"
+---
+
+Cargo's goal is to make modern application package management a core value of
+the Rust programming language.
+
+That concern is driven by a desire to answer questions like:
+
+1. If I want to use `libgit2` in my project, how easy is it?
+2. If I build my project on a different machine, for a different architecture,
+   in CI or for release, am I building from the same source code?
+3. If I build my project for testing, will my indirect dependencies be compiled
+   with debug symbols? If I build my project for release, will my indirect
+   dependencies be compiled with maximum optimizations? How can I be sure?
+4. If someone published a new version of one of my dependencies after I commit,
+   will my CI environment use the same source code? My production environment?
+5. If I add a new dependency (or upgrade one), can that break the build? Can it
+   affect unrelated dependencies? Under what conditions?
+
```

Comment:
My eye had a little trouble tracking the questions above.  Is there a tighter format we could use to capture the reader's eye early?